src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -77,41 +81,109 @@ export const buildAddressSchema = isMilitaryBaseAddress => {
   const addSchema = fullSchema.definitions.addressSchema;
   if (isMilitaryBaseAddress) return cloneDeep(addSchema);
   const schema = cloneDeep(addSchema);
-  delete schema.properties['view:livesOnMilitaryBase'];
-  delete schema.properties['view:livesOnMilitaryBaseInfo'];
+  delete schema.properties[MILITARY_BASE_PATH];
+  delete schema.properties[MILITARY_BASE_INFO_PATH];
   return schema;
 };
 
+const insertArrayIndex = (key, index) => key.replace('[INDEX]', `[${index}]`);
+
+const getOldFormDataPath = (path, index) => {
+  const indexToSlice = index !== null ? path.indexOf(index) + 1 : 0;
+  return path.slice(indexToSlice);
+};
+
+// Temporary storage for city & state if military base checkbox is toggled more
+// than once. Not ideal, but works since this code isn't inside a React widget
+const savedAddress = {
+  city: '',
+  stateCode: '',
+};
+
+/**
+ * Update form data to remove selected military city & state and restore any
+ * previously set city & state when the "I live on a U.S. military base"
+ * checkbox is unchecked. See va.gov-team/issues/42216 for details
+ * @param {object} oldFormData - Form data prior to interaction change
+ * @param {object} formData - Form data after interaction change
+ * @param {array} path - path to address in form data
+ * @param {number} index - index, if form data array of addresses; also included
+ *  in the path, but added here to make it easier to distinguish between
+ *  addresses not in an array with addresses inside an array
+ * @returns {object} - updated Form data with manipulated mailing address if the
+ * military base checkbox state changes
+ */
+export const updateFormDataAddress = (
+  oldFormData,
+  formData,
+  path,
+  index = null, // this is included in the path, but added as
+) => {
+  let updatedData = formData;
+  const address = get(path, formData, {});
+
+  /*
+   * formData and oldFormData are not guaranteed to have the same shape; formData
+   * will always return the entire data object. See below for details on oldFormData
+   *
+   * In the src/platform/forms-system/src/js/containers/FormPage.jsx, if the
+   * address is inside an array (has a `showPagePerItem` index), oldData is set
+   * to the form data from the array index (see the this.formData() function)
+   * but that may not include the address object, so we're passing in a path as
+   * an array and using `getOldFormDataPath` to find the appropriate path
+   */
+  const oldAddress = get(getOldFormDataPath(path, index), oldFormData, {});
+  let { city, stateCode } = address;
+  // console.log(address, oldAddress, index, city, stateCode);
```

Comment:
Can this console log just be removed?

Reply:
Yeah, I think I commited that and @mottie commented that out for me. I'll push a commit to remove it